static/js/context-menu.js

```diff
@@ -60,7 +62,50 @@ const ContextMenu = {
     switch (e.detail.action) {
       case 'edit':
         this.nameInput.value = e.detail.thingName;
+        this.thingType.innerHTML = '';
         this.thingIcon.style.backgroundImage = `url("${e.detail.thingIcon}")`;
+
+        for (const capability of e.detail.capabilities) {
+          const option = document.createElement('option');
+          option.value = capability;
+
+          if (e.detail.selectedCapability === capability) {
```

Comment:
I'm not sure if this is the place to put it, but one thing I'm noticing is that Lights without a selected capability will have their drop-down default to On/Off Switch instead of Light

Reply:
@hobinjk That's not the default. The capabilities are added to the dropdown in the order they're listed in the thing description. I think it's pretty much impossible for the UI to determine what the default should be for any given device.

For instance, if you have a smart plug with a light, the primary device would actually be the `OnOffSwitch`, not the `Light`, but that's impossible to tell with a flat `@type` array.

Reply:
So I guess if we want `Light` to show up first, we need to add that as the first member of the thing description at the adapter level.

Reply:
Alternatively, we could just sort the capabilities, that way they're at least in a consistent order.

Reply:
I think we should establish a simple hierarchy of capabilities where the ones with more functionality from our UI are preferred. I'm not in favor of saying that web thing developers should sort their exposed capabilities in order of preference since to me that feels very non-obvious.

I think it's reasonable that the gateway's UI as a consumer of thing descriptions would make arbitrary decisions to prefer those capabilities which it can best represent. In the case of a smart plug that advertises both Light and OnOffSwitch, I think it makes sense to default to Light since the thing's developer had some reason to include it.

Reply:
@hobinjk I like that. Pushed.